zebra-chain/src/transaction/unmined.rs

```diff
@@ -443,4 +455,28 @@ impl VerifiedUnminedTx {
 
         cost
     }
+
+    /// When deriving `Arbitrary` for `VerifiedUnminedTx` action fields can be generated
+    /// that are against consensus rules. This function prevent those issues by generating
+    /// new conventional and unpaid actions where rules are considered.
+    #[cfg(any(test, feature = "proptest-impl"))]
+    pub fn fix_arbitrary_generated_action_overflows(&mut self) -> Self {
+        let mut rng = rand::thread_rng();
+
+        // The number of actions can't be greater than 2^16.
+        let conventional_actions = rng.gen_range(0..2 ^ 16);
+
+        // The number of unpaid actions can't be greater than `BLOCK_PRODUCTION_UNPAID_ACTION_LIMIT`.
+        let mut unpaid_actions = rng.gen_range(0..zip317::BLOCK_PRODUCTION_UNPAID_ACTION_LIMIT);
```

Comment:
`..value` means "less than `value`", but the verifier accepts "less than or equal to":
https://github.com/ZcashFoundation/zebra/blob/1e47fe5b2bb1b9579fedffbcf1d830194f753666/zebra-chain/src/transaction/unmined/zip317.rs#L181-L183

```suggestion
        let mut unpaid_actions = rng.gen_range(0..=zip317::BLOCK_PRODUCTION_UNPAID_ACTION_LIMIT);
```